internal/yggdrasil/yggdrasil.go

```diff
@@ -245,6 +245,7 @@ func (h *Handler) updateDeploymentStatuses(oldDeployments []v1alpha1.Deployment,
 				deployment.Phase = v1alpha1.EdgeDeploymentPhase(status.Status)
 				deployment.LastTransitionTime = metav1.Now()
 			}
+			deployment.LastDataUpload = metav1.NewTime(time.Time(status.LastDataUpload))
```

Comment:
Shouldn't the assignment of value for `LastDataUpload` depends on the existence of data-paths for the edge deployment ?

Reply:
There is no need. When the data transfer doesn't take place the `status.LastDataUpload` is set to `time.Time` zero value, which in turn doesn't get stored in the CR.